refined two manila modules (#564)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_decommissioning-rhosp-standalone-ceph-NFS-service.adoc

= Decommissioning the {rhos_prev_long} standalone Ceph NFS service

If your deployment uses CephFS through NFS, you must decommission the {rhos_prev_long}({OpenStackShort}) standalone NFS service. Since future software upgrades do not support the previous NFS service, it is recommended that the decommissioning period is short.

> You can discover the new export locations for your existing shares by querying the Shared File Systems API. To stop using the previous NFS server, you need to unmount and remount the shared file systems on each client. If you are consuming the {rhos_component_storage_file} shares with the {rhos_component_storage_file} CSI plugin for {OpenShift}, you can migrate the shares by scaling down the application pods and scaling them back up. Clients spawning new workloads should not use share exports through the previous NFS service. The {rhos_component_storage_file} no longer communicates with the previous NFS service, and so it cannot apply or alter any export rules on the previous NFS service.

Author: What is the intent of the paragraph that starts with “you can discover”? Are some of these things that customers need to do before they start the decommissioning or are some of them suggestions? E.g.,
* Discover the new export locations for your existing shares by querying the Shared File Systems API.
* To stop using the previous NFS server, you need to unmount and remount the shared file systems on each client.
* If you are consuming the {rhos_component_storage_file} shares with the {rhos_component_storage_file} CSI plugin for {OpenShift}, you can migrate the shares by scaling down the application pods and scaling them back up.

Author: Yes, it helps. I think I will put these bullet points under a "Prerequisites" heading, and format them as a list for easier readability.

.Procedure

. Remove the `cephfs_ganesha_server_ip` option from the `manila-share` service configuration:
+
[NOTE]
This restarts the `manila-share` process and removes the export locations that applied to the previous NFS service from all the shares.
+
[source,yaml]
----

> $ cat << __EOF__ > ~/manila.patch

Author: Does “EOF” need to be italicized? That is, can the underscores be removed "__EOF__"?

Reviewer: the "__" (double underscore) characters before and after the "EOF" is just a convention some people use when writing a file through cat..  when you format this as a code block, it should hopefully not translate as italicizing the "EOF" :) 

But, if its confusing, just having "EOF" as Francesco suggested is enough.

Author: I checked the preview again and _EOF_ does not render as italic, so I think it's okay to leave it as is.

= Decommissioning the {rhos_prev_long} standalone Ceph NFS service

If your deployment uses CephFS through NFS, you must decommission the {rhos_prev_long}({OpenStackShort}) standalone NFS service. Since future software upgrades do not support the previous NFS service, it is recommended that the decommissioning period is short.

.Prerequisites

* You identified the new export locations for your existing shares by querying the Shared File Systems API.
* You unmounted and remounted the shared file systems on each client to stop using the previous NFS server.
* If you are consuming the {rhos_component_storage_file} shares with the {rhos_component_storage_file} CSI plugin for {OpenShift}, you migrated the shares by scaling down the application pods and scaling them back up.

[NOTE]

> Clients that are creating new workloads should not use share exports through the previous NFS service. The {rhos_component_storage_file} no longer communicates with the previous NFS service, and so it cannot apply or alter any export rules on the previous NFS service.

Reviewer 2: ```suggestion
Clients that are creating new workloads cannot use share exports through the previous NFS service. The {rhos_component_storage_file} no longer communicates with the previous NFS service, and cannot apply or alter export rules on the previous NFS service.
```

> 

Author: @fmount @gouthampacha Is the new "Prerequisites" section accurate?

__EOF__

----

. Patch the `OpenStackControlPlane` custom resource:
----
$ oc patch openstackcontrolplane openstack --type=merge --patch-file=~/manila.patch
----

> . Optional: To clean up the standalone `ceph-nfs` service from the {OpenStackShort} control plane nodes, you can disable and delete the Pacemaker resources associated with the service:

Author: I added "Optional" to step 3. Can you confirm that this is true?

Reviewer: This section ("Decommissioning the Standalone Ceph-NFS service") is about getting rid of the service; so we wouldn't want it to be Optional.

We've already clarified above that this step can be done any time after RHOSO 18 is operational, but must not be delayed forever...

__EOF__

----

. Patch the `OpenStackControlPlane` custom resource:
+
----
$ oc patch openstackcontrolplane openstack --type=merge --patch-file=~/<manila.patch>
----
* Replace `<manila.patch>` with the name of your patch file.

. Clean up the standalone `ceph-nfs` service from the {OpenStackShort} control plane nodes by disabling and deleting the Pacemaker resources associated with the service:
+
[IMPORTANT]

> You can defer this step until after {rhos_acro} {rhos_curr_ver} is operational. However, in the meantime, you cannot decommission the Controller nodes.

Reviewer 2: ```suggestion
You can defer this step until after {rhos_acro} {rhos_curr_ver} is operational. During this time, you cannot decommission the Controller nodes.
```
I think 'in the meantime' is idiomatic
